### PR TITLE
Modify NSIS options to allow installing Buttercup per-user

### PR DESCRIPTION
**Pull request by Oxalin**
Since Buttercup Desktop will be needed with Chrome's extension (manifest v3), users that were able to install the extension on their local account may end up being unable to install Buttercup Desktop if they don't have administrative privileges.

Adding oneClick:false and changing perMachine:false will launch an installation prompt giving the user a choice to install Buttercup per-user or per-machine (system wide).

Fixes [issue 1247](https://github.com/buttercup/buttercup-desktop/issues/1247)

Tested locally without any visible regression.

### CODE REVIEW

File: source/main/actions/config.ts [hunk 1]
```diff
@@ -6,18 +6,39 @@ import { getLanguage } from "../../shared/library/i18n";
...
+        }`
+    );
+
+    logInfo(` - Start in background: ${preferences.startInBackground ? "Enabled" : "Disabled"}`);
+
+    logInfo(
+        ` - Start with session launch: ${preferences.startWithSession ? "Enabled" : "Disabled"}`
+    );
+    await setStartWithSession(preferences.startWithSession);
+
```
Comment (perry-mitchell) [inline source/main/actions/config.ts:40]: Would you mind removing the extra lines here?

File: source/main/services/config.ts [hunk 2]
```diff
@@ -89,3 +90,34 @@ export async function setVaultSettings(
...
+    const storage = getConfigStorage();
+    const preferences = await storage.getValue("preferences");
+    if (typeof preferences === "undefined") {
+        return false;
+    }
+    return preferences.startInBackground;
+}
+
+export async function setStartWithSession(enable: boolean): Promise<void> {
+    var AutoLaunch = require("auto-launch");
```
Comment (perry-mitchell) [inline source/main/services/config.ts:104]: Why the require here? Can this just be imported?

Also, var => const..

File: source/main/services/config.ts [hunk 3]
```diff
@@ -89,3 +90,34 @@ export async function setVaultSettings(
...
+
+    var autoLauncher = new AutoLaunch({
```
Comment (perry-mitchell) [inline source/main/services/config.ts:106]: var => const :)

**Review by perry-mitchell — CHANGES_REQUESTED** [review]
Some small items. Otherwise looks great!

File: source/main/services/config.ts (hunk 2 above)
Comment (Oxalin) [reply source/main/services/config.ts]: This is fixed by https://github.com/Oxalin/buttercup-desktop/commit/fd7ff2e9ed0da0bfa34ea124ab1eb122383dd6c9

File: source/main/services/config.ts (hunk 3 above)
Comment (Oxalin) [reply source/main/services/config.ts]: Also fixed by https://github.com/Oxalin/buttercup-desktop/commit/fd7ff2e9ed0da0bfa34ea124ab1eb122383dd6c9

**Review by perry-mitchell — APPROVED** [review]

**Comment by perry-mitchell** [comment]
This is great!

image

**Pull request merged**